Documentation / Elasticsearch query endpoint - query samples (#7722)
geonetwork/core-geonetwork: docs/manual/docs/api/search.md

Reference

-   [Search API](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-search.html) ( Elasticsearch Guide )
-   [Multi search API](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-multi-search.html) (Elasticsearch Guide)

## Search API examples

This section provides some query `POST` examples of `/srv/api/search/records/_search` end-point:

1.  To test examples navigate Swagger API documentation at `/srv/api/index.html`

2.  Locate the *search* heading, and the `/search/records/_search` `POST` end-point

3.  Use the *Try it out* button with:
     
     * **bucket**: `metadata`
     * **relatedType**: 
     * **Request body**: Chosen from the examples below

4.  Press **Execute** to run the example.

     ![](img/swagger-search-endpoint.png)

### Text search query

Query with any field for metadata containing the string `infrastructure`, using a query with Lucene syntax and excluding metadata templates:

```json
{
  "query": {
    "bool": {
      "must": [
        {
          "query_string": {

>             "query": "+anytext: infraestructure "

Reviewer: ```suggestion
            "query": "+anytext:infrastructure "
```

          }
        }
      ],
      "filter": [
        {
          "term": {
            "isTemplate": {
              "value": "n"
            }
          }
        }
      ]
    }
  }
}
```

### Subset results

Query with any field for metadata containing the string `infrastructure`, using a query with Lucene syntax and excluding metadata templates, returning a subset of the information:

```json
{
  "query": {
    "bool": {
      "must": [
        {
          "query_string": {

>             "query": "+anytext: infraestructure "

Reviewer: ```suggestion
            "query": "+anytext:infrastructure "
```

          }
        }
      ],
      "filter": [
        {
          "term": {
            "isTemplate": {
              "value": "n"
            }
          }
        }
      ]
    }
  },
  "_source": {
    "includes": [
      "uuid",
      "id",
      "resourceType",
      "resourceTitle*",
      "resourceAbstract*"
    ]
  }
}
```

### Dataset query

Query datasets with title containing the string `infrastructure`, using a query with Lucene syntax and excluding metadata templates:

```json
{
  "query": {
    "bool": {
      "must": [
        {
          "query_string": {

>             "query": "+anytext: infraestructure +resourceType:dataset"

Reviewer: ```suggestion
            "query": "+anytext:infrastructure +resourceType:dataset"
```

## Elasticsearch 

GeoNetwork provides a full Elasticsearch end-point: `/srv/api/search/records/_search` that requires http `POST`, 
accepting a payload with an Elasticsearch query.

Reference

-   [Search API](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-search.html) (Elasticsearch)

### Query examples

> This section provides some query examples. To test them a `POST` request should be sent to the search end-point `/srv/api/search/records/_search` :

Reviewer 2: ```suggestion
This section provides some query `POST` examples of `/srv/api/search/records/_search` end-point.

1.  To test examples navigate Swagger API documentation at `/srv/api/index.html`

2.  Locate the *search* heading, and the `/search/records/_search` `POST` end-point

3.  Use the *Try it out* button with:
     
     * **bucket**: `metadata`
     * ** relatedType**: 
     * **Request body**: Chosen from the examples below
     
     Press **Execute** to run the example.
```
